webapp/src/pages/EditUserPasswordPage/EditUserPasswordPage.tsx

```diff
@@ -0,0 +1,161 @@
+import { useAuth } from '../../shared/hooks/UseAuth';
+
+export default function EditUserPasswordPage() {
+  const { username } = useParams();
+  const { authUser, isLoading } = useAuth(username);
+  const { navigate } = useNavigation();
+  const { warn, notify } = useNotificationContext();
+
+  const initialFormValues: UpdateUserDto = {
+    oldPassword: '',
+    password: '',
+    confirmationPassword: '',
+  };
+  const [formValues, setFormValues] =
+    useState<UpdateUserDto>(initialFormValues);
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = e.target;
+    setFormValues((previousValues: any) => {
+      return { ...previousValues, [name]: value };
+    });
+  };
+
+  function hasValidFormValues() {
+    if (formValues.oldPassword === '') {
+      warn('Old password can not be empty');
+      return false;
+    }
+    if (formValues.password === '') {
+      warn('Password can not be empty');
+      return false;
+    }
+    if (formValues.password !== formValues.confirmationPassword) {
+      warn('Password and Confirmation Password must match');
+      return false;
+    }
+    return true;
+  }
+
+  const handleRequestError = async (error: unknown) => {
+    let errMessage = '';
+    if (error instanceof ResponseError) {
+      const responseBody = await error.response.json();
+      if (responseBody.message) {
+        errMessage = responseBody.message;
+      } else {
+        errMessage = error.response.statusText;
+      }
+    } else if (error instanceof Error) {
+      errMessage = error.message;
+    }
+    warn(errMessage);
+  };
+
+  async function updatePassword() {
+    if (!authUser || !hasValidFormValues()) {
+      return;
+    }
+    try {
+      await usersApi.userControllerUpdateCurrentUser({
+        updateUserDto: formValues,
+      });
+      notify('Password successfully updated');
+      navigate(USER_ME_URL);
+    } catch (error) {
+      handleRequestError(error);
+    }
+  }
+
+  const handleOnSubmit = (e: React.FormEvent<HTMLFormElement>): void => {
+    e.preventDefault();
+    updatePassword().catch((e) => console.error(e));
+  };
+
+  if (isLoading) {
+    return (
+      <div className="edit-user-password">
+        <div className="edit-user-password-loading">
+          <Loading />
+        </div>
+      </div>
+    );
+  }
+
+  if (!authUser) {
+    return <NotFoundPage />;
+  }
+
+  return (
+    <div className="edit-user-password-page">
+      <AvatarPageTemplate
+        isLoading={isLoading}
+        title={`edit/password`}
+        avatarProps={{
+          url: `${AVATAR_EP_URL}/${authUser.avatarId}`,
+          editUrl: `${USER_URL}/${authUser.id}/edit/avatar`,
+          XCoordinate: authUser?.avatarX,
+          YCoordinate: authUser?.avatarY,
+        }}
+        button={{
+          children: 'save',
+          variant: ButtonVariant.SUBMIT,
+          form: 'edit-user-password-form',
+        }}
+        isNotFound={false}
```

Comment:
if you pass here `isNotFound={authUser == null}` I think you don't need this block:
```
  if (!authUser) {
    return <NotFoundPage />;
  }
```
in lines 100 to 103

Reply:
Done